[code_review] AssertionError: Found two functions with the same name

```
Found two functions with the same name
  File "/Users/smujahid/repos/mozilla/bugbug/bugbug/code_search/mozilla.py", line 70, in get_function
    assert found_perfect is None, "Found two functions with the same name"
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/smujahid/repos/mozilla/bugbug/bugbug/code_search/mozilla.py", line 86, in get_function_by_name
    return self.get_function(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/smujahid/repos/mozilla/bugbug/bugbug/tools/code_review.py", line 804, in request_for_function_declarations
    definitions = function_search.get_function_by_name(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/smujahid/repos/mozilla/bugbug/bugbug/tools/code_review.py", line 1050, in run (Current frame)
    requested_functions = request_for_function_declarations(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/smujahid/repos/mozilla/bugbug/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 478, in __call__
    result = fn(*args, **kwargs)
  File "/Users/smujahid/repos/mozilla/bugbug/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
  File "/Users/smujahid/repos/mozilla/bugbug/scripts/code_review_tool_evaluator.py", line 163, in main
    comments = tool.run(patch)
  File "/Users/smujahid/repos/mozilla/bugbug/scripts/code_review_tool_evaluator.py", line 199, in <module>
    main()
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/runpy.py", line 88, in _run_code
    exec(code, run_globals)
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/runpy.py", line 198, in _run_module_as_main
    return _run_code(code, main_globals, None,
AssertionError: Found two functions with the same name
```

--- a/bugbug/code_search/mozilla.py
+++ b/bugbug/code_search/mozilla.py
@@ -64,16 +64,9 @@ class FunctionSearchMozilla(FunctionSearch):
         # TODO: If the function is in a JS file and we found its definition in both JS and C++ files, use the definition in JS file.
         # TODO: If the function is in a non-test file and we found its definition in both test and non-test files, use the definition in the non-test file (and the opposite for functions in test files)
 
-        found_perfect = None
-        for func in result:
-            if func.name == function_name:
-                assert found_perfect is None, "Found two functions with the same name"
-                found_perfect = func
+        perfect_match = [func for func in result if func.name == function_name]
 
-        if found_perfect is not None:
-            return [found_perfect]
-        else:
-            return result
+        return perfect_match if perfect_match else result
 
     def get_function_by_line(
         self, commit_hash: str, path: str, line: int

